icu4c/source/samples/cal/cal.c

```diff
@@ -1,766 +0,0 @@
-    int printVersion = 0;
-    UBool useLongNames = 0;
-    int optInd = 1;
-    char *arg;
-    int32_t month = -1, year = -1;
-    UErrorCode status = U_ZERO_ERROR;
-    
-    
-    /* parse the options */
-    for(optInd = 1; optInd < argc; ++optInd) {
-        arg = argv[optInd];
-        
-        /* version info */
-        if(strcmp(arg, "-v") == 0 || strcmp(arg, "--version") == 0) {
-            printVersion = 1;
-        }
-        /* usage info */
-        else if(strcmp(arg, "-h") == 0 || strcmp(arg, "--help") == 0) {
-            printUsage = 1;
-        }
-        /* use long day names */
-        else if(strcmp(arg, "-l") == 0 || strcmp(arg, "--long") == 0) {
-            useLongNames = 1;
-        }
-        /* POSIX.1 says all arguments after -- are not options */
-        else if(strcmp(arg, "--") == 0) {
-            /* skip the -- */
-            ++optInd;
-            break;
-        }
-        /* unrecognized option */
-        else if(strncmp(arg, "-", strlen("-")) == 0) {
-            printf("cal: invalid option -- %s\n", arg+1);
-            printUsage = 1;
-        }
-        /* done with options, display cal */
-        else {
-            break;
-        }
-    }
-    
-    /* Get the month and year to display, if specified */
-    if(optInd != argc) {
-        
-        /* Month and year specified */
-        if(argc - optInd == 2) {
-            sscanf(argv[optInd], "%d", (int*)&month);
-            sscanf(argv[optInd + 1], "%d", (int*)&year);
-            
-            /* Make sure the month value is legal */
-            if(month < 0 || month > 12) {
-                printf("icucal: Bad value for month -- %d\n", (int)month);
-                
-                /* Display usage */
-                printUsage = 1;
-            }
-            
-            /* Adjust because months are 0-based */
-            --month;
-        }
-        /* Only year specified */
-        else {
-            sscanf(argv[optInd], "%d", (int*)&year);
-        }
-    }
-    
-    /* print usage info */
-    if(printUsage) {
-        usage();
-        return 0;
-    }
-    
-    /* print version info */
-    if(printVersion) {
-        version();
-        return 0;
-    }
-    
-    /* print the cal */
-    cal(month, year, useLongNames, &status);
-    
-    /* ICU cleanup.  Deallocate any memory ICU may be holding.  */
-    u_cleanup();
-
-    return (U_FAILURE(status) ? 1 : 0);
-}
-
-/* Usage information */
-static void
-usage(void)
-{  
-    puts("Usage: icucal [OPTIONS] [[MONTH] YEAR]");
-    puts("");
-    puts("Options:");
-    puts("  -h, --help        Print this message and exit.");
-    puts("  -v, --version     Print the version number of cal and exit.");
-    puts("  -l, --long        Use long names.");
-    puts("");
-    puts("Arguments (optional):");
-    puts("  MONTH             An integer (1-12) indicating the month to display");
-    puts("  YEAR              An integer indicating the year to display");
```

Comment:
@srl295 this one is fairly elaborate and interesting. And most of this C API hasn't changed much, right? Worth something?

Reply:
Let's please keep this one for now.